lib/src/components/NxList/__tests__/NxList.test.tsx

```diff
@@ -0,0 +1,206 @@
+    expect(emptyListComponent.find('span')).toHaveText('Empty message');
+  });
+
+  it('does not show the emptyMessage when there are children', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message">
+        <NxList.Item>
+          <NxList.Text>Foo</NxList.Text>
+        </NxList.Item>
+      </NxList>
+    );
+
+    expect(component).not.toHaveText('Empty message');
+  });
+
+  it('does not show the emptyMessage when isLoading', async function() {
+    const component = await mountAttached(<NxList emptyMessage="Empty message" isLoading />);
+
+    expect(component).not.toHaveText('Empty message');
+  });
+
+  it('does not show the emptyMessage when in error', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message" error="Errr" retryHandler={() => {}}>
+      </NxList>
+    );
+
+    expect(component).not.toHaveText('Empty message');
+  });
+
+  it('removes the emptyMessage when children are added', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message"></NxList>
+    );
+
+    await act(async () => {
+      component.setProps({
+        children: (
+          <>
+            <NxList.Item key="1">
+              <NxList.Text>Foo</NxList.Text>
+            </NxList.Item>
+            <NxList.Item key="2">
+              <NxList.Text>Bar</NxList.Text>
+            </NxList.Item>
+          </>
+        )
+      });
+    });
+    component.update();
+
+    expect(component.find('li').length).toBe(2);
+    expect(component).not.toIncludeText('Empty message');
+  });
+
+  it('renders an empty list', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message"></NxList>
+    );
+
+    expect(component.find('li').length).toBe(1);
+    expect(component).toIncludeText('Empty message');
+  });
+
+  it('shows the loading spinner when isLoading is set', function() {
+    const component = getShallow({isLoading: true});
+
+    expect(component.children()).toMatchSelector('li');
+    expect(component).toContainExactlyOneMatchingElement('li.nx-list__item');
+    expect(component.find('li').children()).toMatchSelector(NxLoadingSpinner);
+  });
+
+  it('shows the error when error is set', async function() {
+    const retryHandler = jest.fn(),
+        component = getShallow({error: 'Error message', retryHandler: retryHandler});
+
+    expect(component.children()).toMatchSelector('li');
+    expect(component).toContainExactlyOneMatchingElement('li.nx-list__item');
+    expect(component.find('li').children()).toMatchSelector(NxLoadError);
+    expect(component.find(NxLoadError)).toHaveProp('error', 'Error message');
+    expect(component.find(NxLoadError)).toHaveProp('retryHandler', retryHandler);
+  });
+
+  it('does not show the emptyMessage when isLoading', async function() {
```

Comment:
These last two seem to duplicate tests already present higher up

Reply:
> These last two seem to duplicate tests already present higher up

The tests have been removed.
55b92e6395a57fc06d88def0b018b36a0a874305